### PR TITLE
Cg 170 edit vessels 

### PR DESCRIPTION
**Pull request by jennikate**
This includes validation

### CODE REVIEW

File: src/components/Forms/EditVessel.jsx [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+
+
+const EditVessel = (props) => {
+  const history = useHistory();
+  const vesselId = props.location.state.vesselId;
+  const [vesselData, setVesselData] = useState();
+  const [formData, setFormData] = useState();
+  const [errors, setErrors] = useState({});
+
+  // Get data to prepopulate the form for this vessel
```
Comment (p3dr0migue1) [inline src/components/Forms/EditVessel.jsx:20]: I would suggest "Populate form using vessel data", "prepopulate" sounds as if the data is populated before loading the form. And "this vessel" which one? ;)

File: src/components/Forms/EditVessel.jsx [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+              case 422: history.push('/sign-in?source=vessels'); break;
+              case 405: history.push('/sign-in?source=vessels'); break;
+              default: history.push('/sign-in?source=vessels'); break;
+            }
+          }
+        });
+    }
+  };
+
+  // Get person data to pass to prepopulate the form
```
Comment (p3dr0migue1) [inline src/components/Forms/EditVessel.jsx:114]: This is slightly confusing, what are you trying to say here? "Send person data to be edited on form"

File: src/components/Forms/EditVessel.jsx [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+          <li className="govuk-breadcrumbs__list-item" aria-current="page">Edit vessel</li>
+        </ol>
+      </div>
+      <main className="govuk-main-wrapper govuk-main-wrapper--auto-spacing" role="main">
+        <div className="govuk-grid-row">
+          <div className="govuk-grid-column-two-thirds">
+            <h1 className="govuk-heading-xl">Edit a vessel</h1>
+            <p className="govuk-body-l">Update the details of the vessel you want to edit.</p>
+            <form id="EditVessel">
+
```
Comment (p3dr0migue1) [inline src/components/Forms/EditVessel.jsx:146]: Lets remove this space please

File: src/components/Forms/validationRules.js [hunk 4]
```diff
@@ -132,3 +132,26 @@ export const dateValidation = (name, value) => {
     default: return null;
   }
 };
+
+export const vesselValidationRules = [
+  {
+    field: 'vesselName',
+    rule: 'required',
+    message: 'You must enter a name',
```
Comment (p3dr0migue1) [inline src/components/Forms/validationRules.js:140]: Please update to 'You must enter a vessel name'

File: src/components/Forms/validationRules.js [hunk 5]
```diff
@@ -132,3 +132,26 @@ export const dateValidation = (name, value) => {
...
+  },
+  {
+    field: 'vesselType',
+    rule: 'required',
+    message: 'You must enter a vessel type',
+  },
+  {
+    field: 'moorings',
+    rule: 'required',
+    message: 'You must enter a usual mooring',
```
Comment (p3dr0migue1) [inline src/components/Forms/validationRules.js:150]: Please update to 'You must enter the vessel usual moorings'

File: src/components/Forms/validationRules.js [hunk 6]
```diff
@@ -132,3 +132,26 @@ export const dateValidation = (name, value) => {
...
+  },
+  {
+    field: 'registration',
+    rule: 'required',
+    message: 'You must enter the registration',
```
Comment (p3dr0migue1) [inline src/components/Forms/validationRules.js:155]: Please update to 'You must enter the vessel registration'

File: src/components/SectionTable.jsx [hunk 7]
```diff
@@ -76,14 +75,20 @@ const SectionTable = ({
                 </tr>
               </thead>
               <tbody className="govuk-table__body">
-                {isPageVessels && data.map((elem) => {
+                {isPageVessels && data.map((vessel) => {
```
Comment (p3dr0migue1) [inline src/components/SectionTable.jsx:78]: :+1:

File: src/components/Forms/EditVessel.jsx (hunk 3 above)
Comment (jennikate) [reply src/components/Forms/EditVessel.jsx:146]: I'll do this on other files when I'm editing them for the other validation work

**Pull request closed without merging**